pkg/stanza/operator/internal/test_common.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/internal"
+
+import (
+	"bufio"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// state is going to keep processing state of the TestReader
+type state struct {
+	ReadFrom  int
+	Processed int
+}
+
+// TestReader is a TestReader which keeps state of readed and processed data
+type TestReader struct {
+	State *state
+	Data  []byte
+}
+
+// NewReader creates TestReader with empty state
+func NewReader(data []byte) TestReader {
```

Comment:
Can we rename this to `NewTestReader`. This method is still exposed internally so it is better to be explicit so we can distinguish source code vs test code